tests/lib/prepare.sh

```diff
@@ -1026,6 +1027,14 @@ uc24_build_initramfs_kernel_snap() {
         cp -a ./extra-kernel-snap/* ./pc-kernel
     fi
 
+    if [ -n "${NESTED_KERNEL_REMOVE_COMPONENTS-}" ]; then
+        #shellcheck disable=SC2016
+        gojq 'del(.components)' --yaml-input pc-kernel/meta/snap.yaml --yaml-output >pc-kernel/meta/snap.yaml.new
+        mv pc-kernel/meta/snap.yaml.new pc-kernel/meta/snap.yaml
```

Comment:
Nitpick both `mv` can be removed with this pattern:

```sh
gojq 'del(.components)' --yaml-input pc-kernel/meta/snap.yaml --yaml-output >pc-kernel/meta/snap.yaml.temp
gojq 'del(.slots)' --yaml-input pc-kernel/meta/snap.yaml.temp --yaml-output >pc-kernel/meta/snap.yaml

Reply:
Sure. But I prefer to make it easier to remove one of them. I think it will be confusing for anybody trying to change that.